### ISSUE TITLE
[Bug]: `aws-privateca-issuer` doesn't works with `csi-cert-manager` for mTLS.

### ISSUE THREAD

**Issue by ankurpshah** [cert-manager/aws-privateca-issuer#69]
### Describe the expected outcome

`aws-privateca-issuer` shold work with `csi-cert-manager` for mTLS without any issuse.

### Describe the actual outcome

`aws-privateca-issuer` is not working with `csi-cert-manager` for mTLS and showing following error:

### Steps to reproduce

Issuer.yaml

```yml
apiVersion: awspca.cert-manager.io/v1beta1
kind: AWSPCAClusterIssuer
metadata:
  name: demo-test-root-ca
spec:
  arn: <CA_ARN>
  region: us-east-1
```

node_server.yaml

```yml
---
apiVersion: v1
kind: Service
metadata:
  labels: {app: hello-mtls-aws-pca}
  name: hello-mtls-aws-pca
  namespace: default
spec:
  type: ClusterIP
  ports:
  - port: 443
    targetPort: 443
  selector: {app: hello-mtls-aws-pca}
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: hello-mtls-aws-pca
  labels: {app: hello-mtls-aws-pca}
  namespace: default
spec:
  replicas: 1
  selector: {matchLabels: {app: hello-mtls-aws-pca}}
  template:
    metadata:
      labels: {app: hello-mtls-aws-pca}
    spec:
      containers:
      - name: hello-mtls-aws-pca
        image: test/cert-manager-server-node:latest
        imagePullPolicy: IfNotPresent
        resources: {requests: {cpu: 10m, memory: 20Mi}}
        volumeMounts:
        - mountPath: "/tls"
          name: tls
      volumes:
      - name: tls
        csi:
          readOnly: true
          driver: csi.cert-manager.io
          volumeAttributes:
            csi.cert-manager.io/issuer-kind: AWSPCAClusterIssuer
            csi.cert-manager.io/issuer-name: demo-test-root-ca
            csi.cert-manager.io/issuer-group: awspca.cert-manager.io
            csi.cert-manager.io/dns-names: hello-mtls-aws-pca.default.svc.cluster.local
            csi.cert-manager.io/duration: "20h"
            csi.cert-manager.io/renew-before: "8h"
```

Create `AWSPCAClusterIssuer` and `Deployment` using above files. Wait for deployment pod coming up. It is waiting for mounting volume and volume mount failing due to no validity mentioned in CertificateRequest.

### Relevant log output

```shell
{"level":"error","ts":1636429708.7008724,"logger":"controllers.CertificateRequest","msg":"failed to request certificate from PCA","certificaterequest":"default/db9e779d-10f6-467b-acc2-341bcd30dd74","error":"operation error ACM PCA: IssueCertificate, https response error StatusCode: 400, RequestID: 07a63afe-f788-4047-8dbe-3857c1557c61, api error ValidationException: 1 validation error detected: Value '0' at 'validity.value' failed to satisfy constraint: Member must have value greater than or equal to 1","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\ngithub.com/cert-manager/aws-privateca-issuer/pkg/controllers.(*CertificateRequestReconciler).Reconcile\n\t/workspace/pkg/controllers/certificaterequest_controller.go:171\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99"}
```
```
E1109 04:22:06.405118       1 manager.go:506] manager "msg"="Failed to issue certificate, retrying after applying exponential backoff" "error"="waiting for request: request \"8961b4c0-3f33-4b30-bbdb-43dedae0a51a\" has failed: failed to request certificate from PCA: operation error ACM PCA: IssueCertificate, https response error StatusCode: 400, RequestID: c6c1fdb3-9a12-424f-bef4-245b7af6c249, api error ValidationException: 1 validation error detected: Value '0' at 'validity.value' failed to satisfy constraint: Member must have value greater than or equal to 1" "volume_id"="csi-fbd8716d6f985bae4fefa59ee13848f66cc8608bb4e4563ae69394424940f6b4"
```
```


### Version

0.3.1 (Latest)

### Have you tried the following?

- [X] Check the [Troubleshooting section](../#troubleshooting)
- [X] Search open [issues](https://github.com/cert-manager/aws-privateca-issuer/issues)

### Category

Supported Workflow Broken

### Severity

Severity 1

**Comment by bmsiegel** [comment]
Thank you for raising this issue with the AWS Private CA Issue plugin. We will review your submission and respond back to you here as soon as possible

**Comment by bmsiegel** [comment]
It looks like this issue is due to the fact that we currently only support using the ValidityDays type. While I work on a fix for you, you can unblock yourself by changing `csi.cert-manager.io/duration` to 24h or greater.

**Comment by anbaig** [comment]
This issue was resolved in release [v1.0.0](https://github.com/cert-manager/aws-privateca-issuer/releases/tag/v1.0.0). If you continue having issues with this after upgrading, please cut a new issue.
